catkit2/services/oceanoptics_spectrometer/oceanoptics_spectrometer.py

```diff
@@ -0,0 +1,187 @@
+        '''
+        super().__init__('oceanoptics_spectrometer')
+
+        self.serial_number = None
+        self.spectrometer = None
+        self.model = None
+        self.pixels_number = None
+
+        self.wavelengths = None
+        self.spectra = None
+
+        self._exposure_time = None
+
+        self.should_be_acquiring = threading.Event()
+        self.should_be_acquiring.set()
+
+    def open(self):
+        '''
+        Open the service.
+
+        This function is called when the service is opened.
+        It initializes the spectrometer and creates the data streams and properties.
+
+        Raises
+        ------
+        RuntimeError
+            If the spectrometer cannot be found.
+        '''
+
+        # Find the spectrometer
+        self.serial_number = str(self.config['serial_number'])
+        try:
+            self.spectrometer = Spectrometer.from_serial_number(self.serial_number)
+        except SeaBreezeError:
+            raise ImportError(f'OceanOptics: Could not find spectrometer with serial number {self.serial_number}')
+
+        # exctract attributes
+        self.model = self.spectrometer.model
+        self.pixels_number = self.spectrometer.pixels
+        self.wavelengths = self.spectrometer.wavelengths()
+
+        # Define and set defaut exposure time
+        self._exposure_time = self.config.get('exposure_time', 1000)
+        self.exposure_time(self._exposure_time)
+
+        # Create datastreams
+        # Use the full sensor size here to always allocate enough shared memory.
+        self.spectra = self.make_data_stream('spectra', 'float32', [self.pixels_number], self.NUM_FRAMES)
+
+    def main(self):
+        '''
+        The main function of the service.
+
+        This function is called when the service is started.
+        '''
+        while not self.should_shut_down:
+            if self.should_be_acquiring.wait(0.05):
+                self.take_one_spectrum()
+                self.end_acquisition()
+
+    def start_acquisition(self):
+        '''
+        Start the acquisition loop.
+        '''
+        self.should_be_acquiring.set()
+
+    def end_acquisition(self):
+        '''
+        End the acquisition loop.
+        '''
+        self.should_be_acquiring.clear()
+
+    def take_one_spectrum(self):
+        '''
+        Measure one spectrum and submit it to the data stream.
+        '''
+        self.spectra.submit_data(self.spectrometer.intensities(), dtype='float32')
+
+    @property
+    def exposure_time(self):
+        '''
+        The exposure time in microseconds.
+
+        Returns:
+        --------
+        int:
+            The exposure time in microseconds.
+        '''
+        return self._exposure_time
+
+    @exposure_time.setter
+    def exposure_time(self, exposure_time: int):
+        '''
+        Set the exposure time in microseconds.
+
+        This property can be used to set the exposure time of the spectrometer.
```

Comment:
I think this line can also go if you want to reduce some lines. It duplicates the first-line comment.